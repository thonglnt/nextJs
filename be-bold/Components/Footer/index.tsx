const Footer = () => {
  return (
    <footer className="mt-5 me-1 ms-1 p-3">
      <div className="row pt-5">
        <div className="col-md-6">
          <h2>Subscribe to our newsletter</h2>
        </div>
        <div className="col-md-6 d-flex">
          <input
            type="text"
            className="form-control"
            name="email"
            placeholder="Enter your email..."
          />
          <span className="input-group-btn">
            <button className="btn btn-dark subscribe" type="button">
              subscribe
            </button>
          </span>
        </div>
      </div>
      <div className="row mt-5 ms-1 me-1">
        <hr />
      </div>
      <div className="row mt-5">
        <div className="col-md-6">
          <img src="/images/logo.png" alt="logo" />
        </div>
        <div className="col-md-2">
          <ul className="list-group list-group-flush list-menu">
            <li className="list-group-item">
              <a href="#">Shop All</a>
            </li>
            <li className="list-group-item">
              <a href="#">Makeup</a>
            </li>
            <li className="list-group-item">
              <a href="#">Skin Care</a>
            </li>
            <li className="list-group-item">
              <a href="#">Hair Care</a>
            </li>
            <li className="list-group-item">
              <a href="#">About</a>
            </li>
            <li className="list-group-item">
              <a href="#">Contact</a>
            </li>
          </ul>
        </div>
        <div className="col-md-2">
          <ul className="list-group list-group-flush list-menu">
            <li className="list-group-item">
              <a href="#">Refund Policy</a>
            </li>
            <li className="list-group-item">
              <a href="#">Terms & Conditions</a>
            </li>
            <li className="list-group-item">
              <a href="#">FAQ</a>
            </li>
            <li className="list-group-item">
              <a href="#">Privacy Policy</a>
            </li>
          </ul>
        </div>
        <div className="col-md-2 social-icon">
          <i className="fa fa-facebook-official me-2" aria-hidden="true"></i>
          <i className="fa fa-twitter me-2" aria-hidden="true"></i>
          <i className="fa fa-instagram" aria-hidden="true"></i>
        </div>
      </div>
      <div className="row mt-5 ms-1 me-1">
        <hr />
      </div>
      <div className="row mt-2">
        <p>Copyright Â© 2023 Be Bold | Powered by Be Bold</p>
      </div>
    </footer>
  );
};

export default Footer;
