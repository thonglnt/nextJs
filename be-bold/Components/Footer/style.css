footer{
  background-color: var(--isabelline);
}

.list-menu li {
  border: none;
  background-color: transparent;
}

.list-menu li a {
  color: var(--black);
  text-decoration: none;
}

.list-menu li:hover a {
  opacity: 0.5;
  color: inherit;
}

.social-icon i:hover {
  color: var(--paleChestnut);
}
