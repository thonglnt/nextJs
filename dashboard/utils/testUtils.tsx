import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@mui/material/styles'

// Themes
import theme from '@self-types/Themes.types'

export const customRender = (children: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
}

export * from '@testing-library/react'
export { customRender as render }
