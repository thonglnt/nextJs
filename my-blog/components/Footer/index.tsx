'use client'

import { Box, Container, Typography, useTheme } from '@mui/material'
import { memo } from 'react'

const Footer = () => {
  const theme = useTheme()
  return (
    <Box component="footer">
      <Box
        padding="75px 0 35px"
        sx={{
          clear: 'both',
        }}
      >
        <Container maxWidth="lg">
          <Box component="aside" margin="0 0 30px" textAlign="center">
            <Typography
              fontSize="13px"
              fontWeight={600}
              data-testid="contact-detail"
            >
              Code your life in an elegant way
            </Typography>
          </Box>
        </Container>
      </Box>
      <Box bgcolor={theme.palette.common.black} textAlign="center">
        <Typography
          data-testid="copyright"
          color={theme.palette.grey[200]}
          padding="13px 0 11px"
          fontSize="12px"
        >
          Â© Copyright 2019-2021 by BeautyOnCode
        </Typography>
      </Box>
    </Box>
  )
}

export default memo(Footer)
