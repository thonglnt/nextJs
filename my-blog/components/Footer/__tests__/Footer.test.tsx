import { render, screen } from '@utils/testUtils'
import Footer from '..'

describe('Footer layouts', () => {
  it('should render Footer layouts', () => {
    const { container } = render(<Footer />)

    expect(container).toMatchSnapshot()

    const contact = screen.getByTestId('contact-detail')
    const copyright = screen.getByTestId('copyright')

    expect(contact).toHaveTextContent('Code your life in an elegant way')
    expect(copyright).toHaveTextContent('Â© Copyright 2019-2021 by BeautyOnCode')
  })
})
