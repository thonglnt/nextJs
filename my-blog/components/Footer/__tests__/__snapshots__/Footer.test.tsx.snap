// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer layouts should render Footer layouts 1`] = `
<div>
  <footer
    class="MuiBox-root css-0"
  >
    <div
      class="MuiBox-root css-1lmv0x8"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      >
        <aside
          class="MuiBox-root css-7y1qy0"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1upvv87-MuiTypography-root"
            data-testid="contact-detail"
          >
            Code your life in an elegant way
          </p>
        </aside>
      </div>
    </div>
    <div
      class="MuiBox-root css-12k0cdt"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-a64yg-MuiTypography-root"
        data-testid="copyright"
      >
        Â© Copyright 2019-2021 by BeautyOnCode
      </p>
    </div>
  </footer>
</div>
`;
